mitodl/open-discussions
# Import ocw-next courses

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3484

#### What's this PR do?

#### How should this be manually tested?
This PR imports ocw next courses into open discussions
Set OCW_NEXT_LIVE_BUCKET=ocw-content-live-qa
Run the migration
Run
```
docker-compose run web ./manage.py backpopulate_ocw_next_data
```
It takes about 30 minutes

Go to ocw-hugo-themes and set
SEARCH_API_URL=http://localhost:8063/api/v0/search/

Verify that you can see studio course in the search.

Also verify that you can import individual courses using

```
docker-compose run web ./manage.py backpopulate_ocw_next_data --course-url-substring 9-s915-developmental-cognitive-neuroscience-spring-2012 --overwrite
```

And that the webhook works by setting OCW_WEBHOOK_KEY in your .env and using Postman to post to

localhost:8063/api/v0/ocw_next_webhook/?webhook_key=<OCW_WEBHOOK_KEY>

with
```
{
    "Records": [
        {
            "s3": {
                "object": {
                    "key": "courses/9-s915-developmental-cognitive-neuroscience-spring-2012/index.html"
                }
            }
        }
    ]
}
```
